### PR TITLE
Update TFM to net7.0

### PR DESCRIPTION
**Pull request by lbussell**
### Problem

Source Build needs all projects to build with a TargetFramework of net7.0 in order to build without pulling in prebuilt binaries.

### Solution

Upgrade dotnet/templating's TargetFramework from net6.0 to net7.0.

### Checks:
- [ ] Added unit tests
- [ ] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style)

### CODE REVIEW

**Comment by lbussell** [comment]
There are many references to `net6.0` in [ParserTests/InstantiateTests.cs](https://github.com/dotnet/templating/blob/main/test/Microsoft.TemplateEngine.Cli.UnitTests/ParserTests/InstantiateTests.cs) but I'm not sure if/how they should be updated or changed.

File: test/dotnet-new3.UnitTests/DotnetNewList.cs [hunk 1]
```diff
@@ -571,8 +571,8 @@ public void CanFallbackToListOption(string command1, string command2)
         [InlineData("foo list bar", "foo", "bar")]
         [InlineData("foo --list bar --language F#", "foo", "bar")]
         [InlineData("foo --list --columns-all bar", "foo", "bar")]
-        [InlineData("foo --list --columns-all --framework net6.0 bar", "bar|net6.0|foo", "--framework")]
-        [InlineData("foo --list --columns-all -other-param --framework net6.0 bar", "bar|--framework|net6.0|foo", "-other-param")]
+        [InlineData("foo --list --columns-all --framework net7.0 bar", "bar|net7.0|foo", "--framework")]
```
Comment (vlada-shubina) [inline test/dotnet-new3.UnitTests/DotnetNewList.cs:574]: this should not be changed

File: test/dotnet-new3.UnitTests/DotnetNewSearch.cs [hunk 2]
```diff
@@ -747,8 +747,8 @@ public void CanFallbackToSearchOption(string command1, string command2)
         [Theory]
         [InlineData("--search foo --columns-all bar", "bar", "foo")]
         [InlineData("--search foo bar", "bar", "foo")]
-        [InlineData("foo --search --columns-all --framework net6.0 bar", "bar|net6.0|foo", "--framework")]
-        [InlineData("foo --search --columns-all -other-param --framework net6.0 bar", "bar|net6.0|--framework|foo", "-other-param")]
+        [InlineData("foo --search --columns-all --framework net7.0 bar", "bar|net7.0|foo", "--framework")]
```
Comment (vlada-shubina) [inline test/dotnet-new3.UnitTests/DotnetNewSearch.cs:750]: this should not be changed

**Review by vlada-shubina — APPROVED** [review]
I don't have any objection to this late change. It doesn't affect NuGet libs, only CLI and tools. 
@JanKrivanek @baronfel thoughts?

**Pull request merged**